combat/CombatTargetting.cpp

```diff
@@ -0,0 +1,150 @@
+#include "CombatTargetting.h"
+
+#include "../universe/Condition.h"
+#include "../universe/Enums.h"
+#include "../universe/Planet.h"
+#include "../universe/Fleet.h"
+#include "../universe/Ship.h"
+#include "../universe/ShipDesign.h"
+#include "../universe/Fighter.h"
+#include "../universe/ValueRef.h"
+
+#include "../util/Logger.h"
+#include "../util/Random.h"
+
+bool Targetting::isPreferredTarget(Targetting::TriggerConditions conditions,
+                                   Targetting::Target target) {
+    const Precision highest_weight = Targetting::findPrecision(conditions);
+    for (long unsigned int i = 0; i < conditions.conditions.size(); i++) {
+        const auto& condition = conditions.conditions[i];
+        if (!condition) {
+            DebugLogger() << "No preferences. Target is perfect." ;
+            return true;
+        }
+
+        if (Targetting::isPreferredTarget(condition, target)) {
+            const int upper = std::round(highest_weight * 1.5f); // HEURISTIC
+            const int improbability = RandInt(0, upper);
+            const int luck = conditions.weights[i];
+            DebugLogger() << "Prioritized locking attempt" <<  [&improbability, &luck, &upper](){
```

Comment:
don't think you need a lambda for this... you're just using `<<` into a `stringstream` but the logger is already a stream.